src/nvim/os/fs.c

```diff
@@ -169,6 +181,37 @@ static bool is_executable_in_path(const char_u *name, char_u **abspath)
       return true;
     }
 
+#ifdef WIN32
+    // Try appending file extensions from $PATHEXT to the name
+    char *buf_end = xstrchrnul((char *)buf, '\0');
+    for (const char *ext = pathext; *ext; ext++) {
+      if (ext[0] != '.' || ext[1] == '\0') {
+        // Skip if there is no suffix after '.' or
+        // if it does not start with '.'
```

Comment:
Ok I think the check should match now.

But isn't this a potential null deref both in the Vim source and here?

```c
 if (p[0] == '.' && (p[1] == '\0' || p[1] == ';'))
```